src/_pyinstaller_hooks_contrib/tests/test_libraries.py

```diff
@@ -27,6 +29,10 @@ def test_jinxed(pyi_builder):
 
 @importorskip('tensorflow')
 def test_tensorflow(pyi_builder):
+    if pyi_builder._mode != 'onedir':
+        pytest.skip('Tensorflow tests support only onedir mode '
+                    'due to potential distribution size.')
+
```

Comment:
We could chuck this into a decorator:
```python
def onedir_only(test):
    def wrapped(pyi_builder):
        if pyi_builder._mode != 'onedir':
            pytest.skip('Tensorflow tests support only onedir mode '
                        'due to potential distribution size.')
        test(pyi_builder)
    return wrapped
```
and apply with `@onedir_only`. Probably not really worth it. What does everyone else think?

Reply:
I think for the decorator, it would make sense to have a generic `onedir_only` decorator with parametrized message in `PyInstaller.utils.tests`. Then, we could use that to create a `tensorflow_onedir_only` decorator with fixed message, and use it with tensorflow tests.

Not sure if anything is gained by using a completely "from-scratch" decorator for just these three tests, though.

Reply:
If this decorator works, please add it in! Sorry that I didn't get round to it before now.

Reply:
Well if we do put it in the main repo then we'll have to wait until the 4.1 release to be able to use it here or the test suite will break running PyInstaller 4.0. I'd say let's just leave this for now. As far as duplicity goes, this case is pretty harmless.